### ISSUE TITLE
get_applied_force() of JoltGeneric6DOFJoint3D not working as expected

### ISSUE THREAD

**Issue by mfg92** [godot-jolt/godot-jolt#993]
Hello everyone,

Thanks for providing this valuable Godot addon. I'm currently working on a small bridge building game, similar to _Bridge It_ or _Poly Bridge_. For the physics I would like to use Jolt, after some testing this seems to be the best fit for my needs. I'm using JoltGeneric6DOFJoint3D to connect the parts of the bridge.
Now I'm struggling with implementing breakable joints and calculating the stress a single element of the bridge is experiencing (for visualisation). My current approach is to use the get_applied_force method of the joints. But it seems to me that this is not working, at least not as I would expect.

I have prepared a demo to illustrate the problem. The red boxes are the anchors created using StaticBody3D. The long yellow boxes are the beams created using RigidBody3D. The adjacent anchors and beams are connected using three JoltGeneric6DOFJoint3D, without any angular DOF constraints, but with either "Linear Limit" **or** "Linear Spring" active on all axes.
I have set Can Sleep to false for the physics nodes.
A Label3D is used to display the value from the get_applied_force method of each joint.

![demo](https://github.com/user-attachments/assets/e8cf2637-d194-4567-9bc9-d15b98eee266)
[demo-project-to-illustrate-the-issue.zip](https://github.com/user-attachments/files/17249494/physics-issue.zip)

For my game I prefer to use the "Linear Spring" variant, but this always gives me 0.0 for the applied force. Is this expected? Are the high values expected in the "Linear Limit" variant and btw what is the unit of this number?

Thanks in advance!

"Linear Limit"  variant   |  "Linear Spring" variant 
:-------------------------:|:-------------------------:
![grafik](https://github.com/user-attachments/assets/9b6a1b47-a88e-4e78-ae59-d3fa18424648) |  ![grafik](https://github.com/user-attachments/assets/4ccd026d-5355-4a1f-91f3-f69d31c89a2b)

**Comment by mihe** [comment]
Sorry for taking so long to respond, it's been a busy week. Unfortunately I'm going to be busy for the next few days as well, so it'll be a while longer before I can take a closer look at this.

**Comment by jrouwe** [comment]
@mihe This looks like it could be my bug. I'll take a look.

**Comment by jrouwe** [comment]
@mfg92 I've figured out what the issue is: The `get_applied_force` call returns the position constraint forces and not the motor forces.

When you call `set_linear_limit_x/y/z_enabled(false)`, you disable the position constraint, so the returned constraint force is zero by definition. When you then call `set_linear_limit_spring_x/y/z_enabled(true)` you're enabling a motor that drives the constraint towards the zero position, but there's currently no way in godot-jolt to get the motor forces.

An easy workaround is to change your script to:

```
extends OptionButton

func _on_item_selected(index: int) -> void:
	var id: int = get_item_id(index)
	print(index)
	var joints = get_tree().get_nodes_in_group("all_joints")
	for j in joints:
		var x: JoltGeneric6DOFJoint3D = j
		if id == 0: # enable linear limits
			x.set_linear_limit_spring_x_enabled(false)
			x.set_linear_limit_spring_y_enabled(false)
			x.set_linear_limit_spring_z_enabled(false)
		elif id == 1: # enable linear springs
			x.set_linear_limit_spring_x_enabled(true)
			x.set_linear_limit_spring_y_enabled(true)
			x.set_linear_limit_spring_z_enabled(true)
```

And in the UI to set the 'Linear Limit Spring' frequency to 10 Hz instead of the 'Linear Spring' frequency:

![image](https://github.com/user-attachments/assets/808dbf70-2ded-4cda-bf6a-0924dadd3947)

This means that the position constraint is active again, so you can read the values with `get_applied_force`.

@mihe This creates the question if `get_applied_force` should also return motor forces or if we need to create a second call (e.g. `get_applied_motor_force`) to obtain them. I've created a PR that does the 1st option (#997). Note that this PR also fixes an issue with the slider and hinge joints that forgot to return some of the constraint lambda. Note that I tested the 6DOF constraint, I didn't have a test case for the other constraints.

**Comment by mfg92** [comment]
@jrouwe Thank you very mutch for pointing out a workaround and providing the fix!

Its confusing to me why the _linear spring_ uses a motor. I tought its an independent group, but probably its an "addition" to the _linear motor_, like the _linear limit spring_ group belongs to the _linear limit_ group. If that's the case a better name would have been to call it _linear motor spring_. But I know that since this name comes from Godot directly, I am addressing the wrong audience with that.

A question that is partially related:
Assuming that there is a beam (RigideBody3D) that is affected by a joint on each of its sides. What is the best way to determine if this body is under compression or tension?
The result of get_applied_forces of both joints should be similar (at least when the system has settled down) as they should cancel each other out. In this scenario, it would be nice to get not only the amount of applied force, but also the direction. If the direction of both applied_forces points away, there is tension in the beam, if they point towards the center of the beam, there is compression.

**Comment by jrouwe** [comment]
> What is the best way to determine if this body is under compression or tension?

Currently there isn't a way to detect that. Jolt returns a 3D vector for the constraint forces, so it should be possible to make the distinction there, but this is not exposed to the godot API (because at the time we didn't see an easy way to convert this vector to a sensible space for a godot API).

What you try is to look at the attachment points of the joint. Because the solver is not perfect, if the system is under tension the blocks would slightly separate, under compression they would slightly penetrate:

![image](https://github.com/user-attachments/assets/d972d35c-508f-48fa-9b95-f2ae9f7ba3cd)

**Comment by mihe** [comment]
This change can now be found in the newly released [0.14.0-stable](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.14.0-stable).

Note that the release on Godot Asset Library is still pending approval, so you might need to download it from GitHub for the time being.
